### ISSUE TITLE
[CNTK] RNN with residuals and masking fails

### ISSUE THREAD

**Issue by javiercorrea** [keras-team/keras#8254]
I was trying to implement a RNN with residuals, and I found out that the model fails to build using CNTK (it does work with tensorflow).

The problem arises when the `mastk_zero` parameter is set to `True` in the following code:
```
from keras.layers import *
from keras.models import Model

model_input = Input(batch_shape=(None,10))
rnn = Embedding(20, 32, mask_zero=True)(model_input)
for i in range(3):
    rnn_ = LSTM(32, return_sequences=True)(rnn)
    # Residual
    rnn = Add()([rnn, rnn_])

model_output = Dense(1)(rnn)
model = Model(model_input, model_output)
```
This is the simplest code I can think of to reproduce the problem.
